tests/test_dle.py

```diff
@@ -610,6 +611,99 @@ def test_incs_no_atomic(self):
+         'Eq(h1[0, y], 3 - y, implicit_dims=(t, x, y))'],
+         2, [3, 2, 1]),
+        (['Eq(y.symbolic_min, g[0, x], implicit_dims=(t, x))',
+          'Eq(y.symbolic_max, g[0, x], implicit_dims=(t, x))',
+          'Eq(h1[0, y], y, implicit_dims=(t, x, y))'],
+         2, [0, 1, 2]),
+        (['Eq(y.symbolic_min, g[0, 0], implicit_dims=(t, x))',
+          'Eq(y.symbolic_max, g[0, 2], implicit_dims=(t, x))',
+          'Eq(h1[0, y], y, implicit_dims=(t, x, y))'],
+         None, [0, 1, 2]),
+        (['Eq(y.symbolic_min, g[0, 2], implicit_dims=(t, x))',
+          'Eq(h1[0, x], y.symbolic_min, implicit_dims=(t, x))'],
+         1, [2, 2, 2]),
+        (['Eq(y.symbolic_min, g[0, 2], implicit_dims=(t, x))',
+          'Eq(h1[0, x], y.symbolic_max, implicit_dims=(t, x))'],
+         1, [2, 2, 2]),
+        (['Eq(y.symbolic_min, g[0, x], implicit_dims=(t, x))',
+          'Eq(y.symbolic_max, g[0, 2], implicit_dims=(t, x))',
+          'Inc(h1[0, y], y, implicit_dims=(t, x, y))'],
+         2, [0, 2, 6]),
+        (['Eq(y.symbolic_min, g[0, x], implicit_dims=(t, x))',
+          'Eq(y.symbolic_max, g[0, 2], implicit_dims=(t, x))',
+          'Inc(h1[0, x], y, implicit_dims=(t, x, y))'],
+         None, [3, 3, 2]),
+        (['Eq(y.symbolic_min, g[0, 0], implicit_dims=(t, x))',
+          'Inc(h1[0, y], x, implicit_dims=(t, x, y))'],
+         2, [3, 3, 3]),
+        (['Eq(y.symbolic_min, g[0, 2], implicit_dims=(t, x))',
+          'Inc(h1[0, x], y.symbolic_min, implicit_dims=(t, x))'],
+         None, [2, 2, 2]),
+        (['Eq(y.symbolic_min, g[0, 2], implicit_dims=(t, x))',
+          'Inc(h1[0, x], y.symbolic_min, implicit_dims=(t, x, y))'],
+         None, [2, 2, 2]),
+        (['Eq(y.symbolic_min, g[0, x], implicit_dims=(t, x))',
+          'Eq(y.symbolic_max, g[0, x]-1, implicit_dims=(t, x))',
+          'Eq(h1[0, y], y, implicit_dims=(t, x, y))'],
+         2, [0, 0, 0])
+    ])
+    def test_edge_cases(self, exprs, simd_level, expected):
+        # Tests for issue #1695
+        t, x, y = dimensions('t x y')
+
+        g = TimeFunction(name='g', shape=(1, 3), dimensions=(t, x),
+                         time_order=0, dtype=np.int32)
+        g.data[0, :] = [0, 1, 2]
+        h1 = TimeFunction(name='h1', shape=(1, 3), dimensions=(t, y), time_order=0)
+        h1.data[0, :] = 0
+
+        # List comprehension would need explicit locals/globals mappings to eval
+        for i, e in enumerate(list(exprs)):
+            exprs[i] = eval(e)
+
+        op = Operator(exprs, opt=('advanced', {'openmp': True}))
+
+        iterations = FindNodes(Iteration).visit(op)
+        assert 'omp for collapse' in iterations[0].pragmas[0].value
+        if simd_level:
+            assert 'omp simd' in iterations[simd_level].pragmas[0].value
+
+        op.apply()
+        assert (h1.data == expected).all()
+
+    def test_simd_space_invariant(self):
+        """
+        Similar to test_space_invariant_v3, testing simd vectorization happens
```

Comment:
I _think_ we could rather add an extra check directly inside `test_space_invariant_v3` and decorate it with a comment along the lines of

```
# NOTE: checks issue #1695
```

instead of a nearly duplicate test? or alternatively, you could trim down `eq` -- I don't think you need all those terms to expose the issue, do you?

if you instead think this current way is better, that's fine then

Reply:
> I _think_ we could rather add an extra check directly inside `test_space_invariant_v3` and decorate it with a comment along the lines of
> 
> ```
> # NOTE: checks issue #1695
> ```
> 
> instead of a nearly duplicate test? or alternatively, you could trim down `eq` -- I don't think you need all those terms to expose the issue, do you?
> 
> if you instead think this current way is better, that's fine then

test_dse does not have a single test for omp or simd checks so I guess we should not diverge for one case.
I trimmed down to a format that still covers the case. Hope it is better now.

Reply:
LGTM